language-tour.md

```diff
@@ -167,31 +179,63 @@ public fun main() {
-  let length = 2.0
-  let measurements = (
-    w: weight,  // error: `weight` is used after escaping here.
-    l: length)
-  print(weight) // use occurs here
-}
-```
-
-The program above is illegal because the value of `weight` *escapes* to initialize another object.
-To better understand, we must keep in mind that Val places great emphasis on the concept of *value independence*.
-
-The constructs of Val manipulate *objects*, which can be thought as independent resources representing some data.
-A binding is a mean to access an object's value, through a name.
+  let base_length = 2.0
+  var length = base_length // <----------------------------------------------------+
+  length += 3.0            //                                                      |
+  let measurements = (     //                                                      |
+    w: weight,             // <-----------------------------------------------+    |
+    l: length)             //                                                 |    |
+  print(weight)            // error: `weight` used after being consumed here -+    |
+                           //                                                      |
+  print(base_length)       // error: `base_length` used after being consumed here -+
+}
+```
+
+The program above is illegal because the values of `weight` and `base_length` are consumed to initialize other objects.
+This design follows from two of Val's core principles:
+1. In a Val program, all **copies are explicit by default**. 
+Languages that copy most values implicitly (C++, Swift, R, …) often do so at great expense to performance, and avoiding implicit copies can itself incur a great expense in code size, in code and language complexity, and in development speed.
+Fortunately, Val naturally needs many fewer copies than other languages, and the Val compiler can offer to insert the copies that you may miss.
+Val also supports selectively-enabled implicit copies (for scripting, education purpose, etc)—but that is not the default.
+2. In a Val program the **values of distinct bindings and distinct objects are *independent* under mutation**. Languages that allow two accessible names to bind to the same mutable object (JavaScript, Python, Ruby, Lua, parts of C++ and Swift) are prone to hidden interactions, race conditions, and easily scale up into systems that can't be documented, tested, or understood.
+
+{% comment %}
+
+I don't think the following description is appropriate here.  I feel I understand Val code perfectly
+well and the idea that an object has exactly one owner at any time is not part of my way of
+thinking.  In fact, that seems like a very complicated way to understand `let x = 1; let y = x;
+...`; you may have to transfer ownership around based on the site of the last use. I think of our
+bindings as being like reader/writer locks: there are either multiple owners through which the
+object is immutable, or one owner through which it is mutable.
+
```

Comment:
Okay, I like this analogy! single-writer/multiple-reader is a very well-established principle, so it might be easier to build on that.